vector/src/main/java/fr/gouv/tchap/media/MediaScanManager.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2018 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.gouv.tchap.media;
+
+import android.support.annotation.Nullable;
+import org.matrix.androidsdk.rest.callback.ApiCallback;
+import org.matrix.androidsdk.rest.model.crypto.EncryptedFileInfo;
+
+import fr.gouv.tchap.model.MediaScan;
+import io.realm.Realm;
+
+public class MediaScanManager {
+
+    private static final String LOG_TAG = MediaScanManager.class.getSimpleName();
+
+    MediaScanDao mMediaScanDao;
+
+    /**
+     * Constructor
+     *
+     * @param realm instance of mediaScanDao.
+     */
+    public MediaScanManager(Realm realm) {
+        mMediaScanDao = new MediaScanDao(realm);
+    }
+
+    /**
+     * Get the current scan result on a media (including antivirus status).
+     * Trigger an antivirus scan if it is not already done and if a callback is provided.
+     *
+     * @param url       the media url.
+     * @param callback  optional async response handler.
+     * @return the current scan status for a dedicated media.
+     */
+    public MediaScan scanMedia(String url, @Nullable ApiCallback<Void> callback) {
```

Comment:
It's strange to return a MediaScan and to trigger network. The `callback` parameter should be a specific listener of a proper type